Blocks: Allow the Default Style selector to be hidden. (#20620)
WordPress/gutenberg: docs/designers-developers/developers/block-api/block-registration.md

Plugins and Themes can also register [custom block style](/docs/designers-developers/developers/filters/block-filters.md#block-style-variations) for existing blocks.

#### allowDefaultStyle (optional)

-   **Type:** `Boolean`

> When the style picker is shown, it may also show a selector to allow the user to select a default style to use for this block type. If you prefer not to show the selector, set this property to `false`.

Reviewer: ```suggestion
When the style picker is shown, a selector is displayed so the user can select a default style for this block type. If you prefer not to show the selector, set this property to `false`, the default value is `true`.
```

alignWide: false,
```

> -   `defaultStyle` (default `true`): When the style picker is shown, a dropdown is displayed so the user can select a default style for this block type. If you prefer not to show the dropdown, set this property to `false`.

Reviewer 2: Noting that the editor setting itself is still marked as experimental:

https://github.com/WordPress/gutenberg/blob/8a94254b6adac030f5a6ffd8e718972e1f19f48e/packages/block-editor/src/components/default-style-picker/index.js#L23

Which leaves this in an awkward mix of semi-stable support.

Related: #16465

@jorgefilipecosta Do you have a sense whether this is something that will plan to be stabilized? Or at least that even if the setting shape itself changes, the desire for "default style" user control will remain?

Author: @jorgefilipecosta: Pinging you on this question: is there any plan to stabilise this feature?

Reviewer 3: We still have a considerable amount of work to do on https://github.com/WordPress/gutenberg/issues/7551. Allowing the user to change the default styles was just a step.
We also want to allow themes to change default styles https://github.com/WordPress/gutenberg/pull/16532.
And we want to allow users to create their block styles from the UI. I think this API should not be stabilized before at least these two follow-ups are merged and tested (as the follow-up work may benefit from a better mechanism to store and pass default styles around the several artifacts we have). So there are no immediate plans for stabilization.

> -   `defaultStyle` (default `true`): When the style picker is shown, a dropdown is displayed so the user can select a default style for this block type. If you prefer not to show the dropdown, set this property to `false`.

Reviewer 2: I wonder if there's a risk of ambiguity between `defaultStyle` here and the `isDefault` property of [a registered block variation](https://developer.wordpress.org/block-editor/developers/filters/block-filters/#block-style-variations). I'm not sure I have any suggestion here, but I do note that this particular support is targeted specifically at the control given to the user in selecting their own default, vs. a default amongst a set of registered variations.

Author: Maybe `defaultStyleDropdown`, or something of that nature? It's wandering a little to close to tying feature settings to feature implementation details, though. This is particularly evident with this feature, since the first iteration of it was as a set of checkboxes, rather than a dropdown.

Reviewer 4: How about `showDefaultStylePicker`?

Author: Since it's a flag in `supports`, the name should read well with that before it. Eg, `defaultStylePicker`, as in "supports default style picker" would potentially work.

Author: Renamed in e769c123f.
